### ISSUE TITLE
Can't connect to China server anymore ?

### ISSUE THREAD

**Issue by Yterz** [And3rsL/Deebot-for-Home-Assistant#158]
**Describe the bug**
After I restarted HA on the 20th July, HA wasn't able to connect to my vacuum anymore, and I get an error in HA log, about `('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer')` (see Logs for detail).
It was still able to the day before, as it sent a command with success.

I tried removing & re-adding the integration, but with no success, integration prompt return blank with "Unknown error" (I refilled it for the screenshot)
<img width="395" alt="Capture d’écran 2021-07-21 à 17 10 23" src="https://user-images.githubusercontent.com/73762135/126512932-e311792b-8842-4d89-b194-b97e58e6ac90.png">
I still managed to connect to an other French Ecovacs account, and get to the configuration step where I can pick devices, therefore I became more suspicious on the "cn" (ie China Mainland server) part of the connection.

That was with Deebot-for-Home-Assistant 2.1.2, and I also tested 3.0.0b12 with no more success (but I have an unrelated migration error now that I reinstalled 2.1.2, and still can't re-add the integration to fix that ...)

**System Informations:**
 - Device: Deebot T8 AIVI+
 - Home Assistant Version: 2021.7.3
 - Hardware: Raspberry Pi 4

**Additional context**
I still can access the vacuum through the EcovacsHome application without issue, and log in and out of my account using the short ID + password.
But browsing in the application other than on the vacuum part is throwing a lot of small error message that it wasn't before : `data submission error,` with `(82015)`, `(80015)`, `(82016)` or `(80020)` (and maybe some other)
Also, the website of ecovacs.cn itself seems to have connection issue, sometime working, some other the browser claim it can't reach it.
That made me even more suspicious on the China Mainland server part of the connection. With the weird part that I still get to access my vacuum through the application without issue, at worst it appear offline for 0,5s at the menu to enter smart cleaning, but once enter I get no issue and is very responsive to instruction. Maybe the application spam the connection until one get through ?

No idea what I could do though, and if it's indeed the issue, and if I'm the only one with that problem...

I'd be willing to PM my Chinese log if it can help someone knowing what to look for

**Logs**
```
Logger: homeassistant.config_entries
Source: custom_components/deebot/hub.py:30 
Integration: Deebot for Home Assistant (documentation) 
First occurred: 13:36:09 (1 occurrences) 
Last logged: 13:36:09

Error setting up entry ECHWV4L5 for deebot
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 382, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1010, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 411, in connect
    self.sock = ssl_wrap_socket(
  File "/usr/local/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 449, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(
  File "/usr/local/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 493, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
  File "/usr/local/lib/python3.9/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/local/lib/python3.9/ssl.py", line 1040, in _create
    self.do_handshake()
  File "/usr/local/lib/python3.9/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
ConnectionResetError: [Errno 104] Connection reset by peer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 439, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 755, in urlopen
    retries = retries.increment(
  File "/usr/local/lib/python3.9/site-packages/urllib3/util/retry.py", line 532, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/usr/local/lib/python3.9/site-packages/urllib3/packages/six.py", line 769, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 382, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1010, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 411, in connect
    self.sock = ssl_wrap_socket(
  File "/usr/local/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 449, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(
  File "/usr/local/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 493, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
  File "/usr/local/lib/python3.9/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/local/lib/python3.9/ssl.py", line 1040, in _create
    self.do_handshake()
  File "/usr/local/lib/python3.9/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 293, in async_setup
    result = await component.async_setup_entry(hass, self)  # type: ignore
  File "/config/custom_components/deebot/__init__.py", line 30, in async_setup_entry
    hass.data[DOMAIN][entry.entry_id] = await hass.async_add_executor_job(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/deebot/hub.py", line 30, in __init__
    self.ecovacs_api = EcoVacsAPI(
  File "/usr/local/lib/python3.9/site-packages/deebotozmo/__init__.py", line 68, in __init__
    login_info = self.__call_main_api(
  File "/usr/local/lib/python3.9/site-packages/deebotozmo/__init__.py", line 134, in __call_main_api
    api_response = requests.get(
  File "/usr/local/lib/python3.9/site-packages/requests/api.py", line 76, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 498, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
```

**Comment by edenhaus** [comment]
First downgrade it not supported. You need to remove the deebot in the integration view and set up it again as there is a config migration between v2->v3.

I don't live in China, nor I have a chinese account. Currently I don't have time to investigate it more...
If someone has time please feel free to investigate/analyze it

**Comment by Yterz** [comment]
Connection working again !

I tested again to add the integration, and today it succeeded without any issue, on 2.1.2, back as before.
Also the application don't have any error message `data submission error` anymore, so I guess it was only an issue on Ecovacs side

Tested 3.0.0, but as announced I had the same problem as #154 , so I'll stay on v2 for now

For intel, unrelated :
Chinese account are needed because the Deebot come from China (ordered from AliExpress), not related to where it's used (France here). The account was provided by the seller, because mandatory linked to a Chinese phone number (so no email, only Ecovacs ID to log).
